pkg/controller/migmigration/postValidate.go

```diff
@@ -0,0 +1,298 @@
+	}
+
+	// Validate Jobs
+	healthy, err := t.validateJobs(client, options)
+	if err != nil {
+		return ValidationFailed, err
+	}
+	if !healthy {
+		return ValidationFailed, nil
+	}
+
+	// Validate pod owner resources
+	// includes ReplicaSets, Deployments, etc.
+	return t.validateOwnerResources(client, options)
+}
+
+func (t *Task) validateOwnerResources(client k8sclient.Client, options *k8sclient.ListOptions) (int, error) {
+	podManagersList := unstructured.UnstructuredList{}
+
+	for _, gvk := range podManagingResources {
+		podManagersList.SetGroupVersionKind(gvk)
+		err := client.List(context.TODO(), options, &podManagersList)
+		if err != nil {
+			return ValidationFailed, err
+		}
+
+		for _, podManager := range podManagersList.Items {
+			expectedReplicas, found, err := unstructured.NestedFieldNoCopy(podManager.Object, "spec", "replicas")
+			if err != nil {
+				return ValidationFailed, err
+			}
+			if !found {
+				err = fmt.Errorf("Replicas field was not found in %s kind", gvk.String())
+				return ValidationFailed, err
+			}
+
+			readyReplicas, found, err := unstructured.NestedFieldNoCopy(podManager.Object, "status", "readyReplicas")
+			if err != nil {
+				return ValidationFailed, err
+			}
+			if found && expectedReplicas == readyReplicas {
+				continue
+			}
+
+			healthy, err := t.checkPodsHealth(client, options)
+			if err != nil || !healthy {
+				return ValidationFailed, err
+			}
+			return ValidationProgressing, nil
+		}
+	}
+	return ValidationSucceeded, nil
+}
+
+func (t *Task) validateJobs(client k8sclient.Client, options *k8sclient.ListOptions) (bool, error) {
+	jobList := batchv1.JobList{}
+
+	err := client.List(context.TODO(), options, &jobList)
+	if err != nil {
+		return false, err
+	}
+
+	for _, job := range jobList.Items {
+		if job.Status.Failed == 0 {
+			continue
+		}
+
+		// Check the state of pods
+		healthy, err := t.checkPodsHealth(client, options)
+		if err != nil {
+			return false, err
+		}
+		if !healthy {
+			return false, err
+		}
+		break
+	}
+	return true, nil
+}
+
+func (t *Task) checkPodsHealth(client k8sclient.Client, options *k8sclient.ListOptions) (bool, error) {
+	failedPodCounter := 0
+	pendingPodCounter := 0
+	podNames := []string{}
+	podList := corev1.PodList{}
+	eventList := corev1.EventList{}
+
+	err := client.List(context.TODO(), options, &podList)
+	if err != nil {
+		return false, err
+	}
+
+	err = client.List(context.TODO(), options, &eventList)
+	if err != nil {
+		return false, err
+	}
+
+	for _, pod := range podList.Items {
+		switch pod.Status.Phase {
+		case corev1.PodSucceeded, corev1.PodRunning:
+			continue
+		case corev1.PodFailed:
+			failedPodCounter++
+		default:
+			pendingPodCounter++
+		}
+
+		podNames = append(podNames, pod.GetName())
+	}
+
+	correlatedEvents := collectRelatedEvents(podNames, eventList.Items)
+	eventsJSON, err := json.MarshalIndent(correlatedEvents, "", " ")
+	if err != nil {
+		return false, err
+	}
+
+	if failedPodCounter > 0 {
```

Comment:
Looks like only the _category_ is different between the `if` and `else`.  Perhaps it would be good to reduce the duplication by:
```
category := Advisory
if failedPodCounter > 0 {
    category = Critical
}
```

Then:  `Category: category`.

Reply:
Thinking this should not be a `Critical` category but rather see `Reason: Failed` below.